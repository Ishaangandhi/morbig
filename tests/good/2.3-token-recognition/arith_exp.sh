echo  $((1 * $((2*3)) * 42))

echo $((x+=`echo 2`))

echo $((n * $(fact $(($n-1)) ) ))

echo "$n"

echo $n

echo $(($n))
