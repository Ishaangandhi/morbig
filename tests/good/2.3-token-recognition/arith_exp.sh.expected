[
  "Program_LineBreak_CompleteCommands_LineBreak",
  [ "LineBreak_Empty" ],
  [
    "CompleteCommands_CompleteCommands_NewlineList_CompleteCommand",
    [
      "CompleteCommands_CompleteCommands_NewlineList_CompleteCommand",
      [
        "CompleteCommands_CompleteCommands_NewlineList_CompleteCommand",
        [
          "CompleteCommands_CompleteCommands_NewlineList_CompleteCommand",
          [
            "CompleteCommands_CompleteCommands_NewlineList_CompleteCommand",
            [
              "CompleteCommands_CompleteCommand",
              [
                "CompleteCommand_CList",
                [
                  "CList_AndOr",
                  [
                    "AndOr_Pipeline",
                    [
                      "Pipeline_PipeSequence",
                      [
                        "PipeSequence_Command",
                        [
                          "Command_SimpleCommand",
                          [
                            "SimpleCommand_CmdName_CmdSuffix",
                            [
                              "CmdName_Word",
                              [ "Word", "echo", [ [ "WordName", "echo" ] ] ]
                            ],
                            [
                              "CmdSuffix_Word",
                              [
                                "Word",
                                "$((1 * $((2*3)) * 42))",
                                [
                                  [
                                    "WordArith",
                                    [
                                      "Word",
                                      "1 * $((2*3)) * 42",
                                      [
                                        [ "WordLiteral", "1 * " ],
                                        [
                                          "WordArith",
                                          [
                                            "Word",
                                            "2*3",
                                            [ [ "WordLiteral", "2*3" ] ]
                                          ]
                                        ],
                                        [ "WordLiteral", " * 42" ]
                                      ]
                                    ]
                                  ]
                                ]
                              ]
                            ]
                          ]
                        ]
                      ]
                    ]
                  ]
                ]
              ]
            ],
            [ "NewLineList_NewLine" ],
            [
              "CompleteCommand_CList",
              [
                "CList_AndOr",
                [
                  "AndOr_Pipeline",
                  [
                    "Pipeline_PipeSequence",
                    [
                      "PipeSequence_Command",
                      [
                        "Command_SimpleCommand",
                        [
                          "SimpleCommand_CmdName_CmdSuffix",
                          [
                            "CmdName_Word",
                            [ "Word", "echo", [ [ "WordName", "echo" ] ] ]
                          ],
                          [
                            "CmdSuffix_Word",
                            [
                              "Word",
                              "$((x+=`echo 2`))",
                              [
                                [
                                  "WordArith",
                                  [
                                    "Word",
                                    "x+=`echo 2`",
                                    [
                                      [ "WordLiteral", "x+=" ],
                                      [
                                        "WordSubshell",
                                        [ "SubShellKindBackQuote" ],
                                        [
                                          "Program_LineBreak_CompleteCommands_LineBreak",
                                          [ "LineBreak_Empty" ],
                                          [
                                            "CompleteCommands_CompleteCommand",
                                            [
                                              "CompleteCommand_CList",
                                              [
                                                "CList_AndOr",
                                                [
                                                  "AndOr_Pipeline",
                                                  [
                                                    "Pipeline_PipeSequence",
                                                    [
                                                      "PipeSequence_Command",
                                                      [
                                                        "Command_SimpleCommand",
                                                        [
                                                          "SimpleCommand_CmdName_CmdSuffix",
                                                          [
                                                            "CmdName_Word",
                                                            [
                                                              "Word",
                                                              "echo",
                                                              [
                                                                [
                                                                  "WordName",
                                                                  "echo"
                                                                ]
                                                              ]
                                                            ]
                                                          ],
                                                          [
                                                            "CmdSuffix_Word",
                                                            [
                                                              "Word",
                                                              "2",
                                                              [
                                                                [
                                                                  "WordLiteral",
                                                                  "2"
                                                                ]
                                                              ]
                                                            ]
                                                          ]
                                                        ]
                                                      ]
                                                    ]
                                                  ]
                                                ]
                                              ]
                                            ]
                                          ],
                                          [ "LineBreak_Empty" ]
                                        ]
                                      ]
                                    ]
                                  ]
                                ]
                              ]
                            ]
                          ]
                        ]
                      ]
                    ]
                  ]
                ]
              ]
            ]
          ],
          [ "NewLineList_NewLine" ],
          [
            "CompleteCommand_CList",
            [
              "CList_AndOr",
              [
                "AndOr_Pipeline",
                [
                  "Pipeline_PipeSequence",
                  [
                    "PipeSequence_Command",
                    [
                      "Command_SimpleCommand",
                      [
                        "SimpleCommand_CmdName_CmdSuffix",
                        [
                          "CmdName_Word",
                          [ "Word", "echo", [ [ "WordName", "echo" ] ] ]
                        ],
                        [
                          "CmdSuffix_Word",
                          [
                            "Word",
                            "$((n * $(fact $(($n-1)) ) ))",
                            [
                              [
                                "WordArith",
                                [
                                  "Word",
                                  "n * $(fact $(($n-1)) ) ",
                                  [
                                    [ "WordLiteral", "n * " ],
                                    [
                                      "WordSubshell",
                                      [ "SubShellKindParentheses" ],
                                      [
                                        "Program_LineBreak_CompleteCommands_LineBreak",
                                        [ "LineBreak_Empty" ],
                                        [
                                          "CompleteCommands_CompleteCommand",
                                          [
                                            "CompleteCommand_CList",
                                            [
                                              "CList_AndOr",
                                              [
                                                "AndOr_Pipeline",
                                                [
                                                  "Pipeline_PipeSequence",
                                                  [
                                                    "PipeSequence_Command",
                                                    [
                                                      "Command_SimpleCommand",
                                                      [
                                                        "SimpleCommand_CmdName_CmdSuffix",
                                                        [
                                                          "CmdName_Word",
                                                          [
                                                            "Word",
                                                            "fact",
                                                            [
                                                              [
                                                                "WordName",
                                                                "fact"
                                                              ]
                                                            ]
                                                          ]
                                                        ],
                                                        [
                                                          "CmdSuffix_Word",
                                                          [
                                                            "Word",
                                                            "$(($n-1))",
                                                            [
                                                              [
                                                                "WordArith",
                                                                [
                                                                  "Word",
                                                                  "$n-1",
                                                                  [
                                                                    [
                                                                    "WordVariable",
                                                                    [
                                                                    "VariableAtom",
                                                                    "n",
                                                                    [
                                                                    "NoAttribute"
                                                                    ]
                                                                    ]
                                                                    ],
                                                                    [
                                                                    "WordLiteral",
                                                                    "-1"
                                                                    ]
                                                                  ]
                                                                ]
                                                              ]
                                                            ]
                                                          ]
                                                        ]
                                                      ]
                                                    ]
                                                  ]
                                                ]
                                              ]
                                            ]
                                          ]
                                        ],
                                        [ "LineBreak_Empty" ]
                                      ]
                                    ],
                                    [ "WordLiteral", " " ]
                                  ]
                                ]
                              ]
                            ]
                          ]
                        ]
                      ]
                    ]
                  ]
                ]
              ]
            ]
          ]
        ],
        [ "NewLineList_NewLine" ],
        [
          "CompleteCommand_CList",
          [
            "CList_AndOr",
            [
              "AndOr_Pipeline",
              [
                "Pipeline_PipeSequence",
                [
                  "PipeSequence_Command",
                  [
                    "Command_SimpleCommand",
                    [
                      "SimpleCommand_CmdName_CmdSuffix",
                      [
                        "CmdName_Word",
                        [ "Word", "echo", [ [ "WordName", "echo" ] ] ]
                      ],
                      [
                        "CmdSuffix_Word",
                        [
                          "Word",
                          "\"$n\"",
                          [
                            [
                              "WordDoubleQuoted",
                              [
                                "Word",
                                "$n",
                                [
                                  [
                                    "WordVariable",
                                    [
                                      "VariableAtom",
                                      "n",
                                      [ "NoAttribute" ]
                                    ]
                                  ]
                                ]
                              ]
                            ]
                          ]
                        ]
                      ]
                    ]
                  ]
                ]
              ]
            ]
          ]
        ]
      ],
      [ "NewLineList_NewLine" ],
      [
        "CompleteCommand_CList",
        [
          "CList_AndOr",
          [
            "AndOr_Pipeline",
            [
              "Pipeline_PipeSequence",
              [
                "PipeSequence_Command",
                [
                  "Command_SimpleCommand",
                  [
                    "SimpleCommand_CmdName_CmdSuffix",
                    [
                      "CmdName_Word",
                      [ "Word", "echo", [ [ "WordName", "echo" ] ] ]
                    ],
                    [
                      "CmdSuffix_Word",
                      [
                        "Word",
                        "$n",
                        [
                          [
                            "WordVariable",
                            [ "VariableAtom", "n", [ "NoAttribute" ] ]
                          ]
                        ]
                      ]
                    ]
                  ]
                ]
              ]
            ]
          ]
        ]
      ]
    ],
    [ "NewLineList_NewLine" ],
    [
      "CompleteCommand_CList",
      [
        "CList_AndOr",
        [
          "AndOr_Pipeline",
          [
            "Pipeline_PipeSequence",
            [
              "PipeSequence_Command",
              [
                "Command_SimpleCommand",
                [
                  "SimpleCommand_CmdName_CmdSuffix",
                  [
                    "CmdName_Word",
                    [ "Word", "echo", [ [ "WordName", "echo" ] ] ]
                  ],
                  [
                    "CmdSuffix_Word",
                    [
                      "Word",
                      "$(($n))",
                      [
                        [
                          "WordArith",
                          [
                            "Word",
                            "$n",
                            [
                              [
                                "WordVariable",
                                [ "VariableAtom", "n", [ "NoAttribute" ] ]
                              ]
                            ]
                          ]
                        ]
                      ]
                    ]
                  ]
                ]
              ]
            ]
          ]
        ]
      ]
    ]
  ],
  [ "LineBreak_Empty" ]
]